applications/dashboard/models/class.spammodel.php

```diff
@@ -110,9 +110,99 @@ public static function isSpam($RecordType, $Data, $Options = array()) {
                     break;
             }
 
-            LogModel::insert('Spam', $RecordType, $Data, $LogOptions);
+            // If this is a discussion or a comment, it needs some special handling.
+            if ($RecordType == 'Comment' || $RecordType == 'Discussion') {
+                // Grab the record ID, if available.
```

Comment:
What do you think of collapsing this entire `switch` into just `$recordID = intval(val($RecordType.'ID', $Data, 0));` ?